File: VB/MvcTreeListDragDrop/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports MvcTreeListDragDrop.Model

Namespace MvcTreeListDragDrop.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function
		Public Function LeftTreePartial() As ActionResult
			Return PartialView("LeftTreePartial")
		End Function
		Public Function RightTreePartial() As ActionResult
			Return PartialView("RightTreePartial")
		End Function
		Public Function UpdateTreeListContainer(ByVal sourceKey As String, ByVal newParentKey As String) As ActionResult
			MyProvider.MoveNode(sourceKey, newParentKey)
			Return PartialView("TreeListContainerPartial")
		End Function
	End Class
End Namespace

File: CS/WebApplication1/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using WebApplication1.Model;

namespace WebApplication1.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        public ActionResult LeftTreePartial() {
            return PartialView("LeftTreePartial");
        }
        public ActionResult RightTreePartial() {
            return PartialView("RightTreePartial");
        }
        public ActionResult UpdateTreeListContainer(string sourceKey, string newParentKey) {
            MyProvider.MoveNode(sourceKey, newParentKey);
            return PartialView("TreeListContainerPartial");
        }
    }
}